internal/command/command_plugin.go

```diff
@@ -38,75 +40,95 @@ type (
-		commandService   commandService
-		subscribeChannel chan *mpi.ManagementPlaneRequest
-		subscribeMutex   sync.Mutex
+		messagePipe       bus.MessagePipeInterface
+		config            *config.Config
+		subscribeCancel   context.CancelFunc
+		conn              grpc.GrpcConnectionInterface
+		commandService    commandService
+		subscribeChannel  chan *mpi.ManagementPlaneRequest
+		commandServerType model.ServerType
+		subscribeMutex    sync.Mutex
 	}
 )
 
-func NewCommandPlugin(agentConfig *config.Config, grpcConnection grpc.GrpcConnectionInterface) *CommandPlugin {
+func NewCommandPlugin(agentConfig *config.Config, grpcConnection grpc.GrpcConnectionInterface,
+	commandServerType model.ServerType,
+) *CommandPlugin {
 	return &CommandPlugin{
-		config:           agentConfig,
-		conn:             grpcConnection,
-		subscribeChannel: make(chan *mpi.ManagementPlaneRequest),
+		config:            agentConfig,
+		conn:              grpcConnection,
+		subscribeChannel:  make(chan *mpi.ManagementPlaneRequest),
+		commandServerType: commandServerType,
 	}
 }
 
 func (cp *CommandPlugin) Init(ctx context.Context, messagePipe bus.MessagePipeInterface) error {
-	slog.DebugContext(ctx, "Starting command plugin")
+	newCtx := context.WithValue(
+		ctx,
+		logger.ServerTypeContextKey, slog.Any(logger.ServerTypeKey, cp.commandServerType.String()),
+	)
+	slog.DebugContext(newCtx, "Starting command plugin")
 
 	cp.messagePipe = messagePipe
 	cp.commandService = NewCommandService(cp.conn.CommandServiceClient(), cp.config, cp.subscribeChannel)
 
-	go cp.monitorSubscribeChannel(ctx)
+	go cp.monitorSubscribeChannel(newCtx)
 
 	return nil
 }
 
 func (cp *CommandPlugin) Close(ctx context.Context) error {
-	slog.InfoContext(ctx, "Closing command plugin")
+	newCtx := context.WithValue(
+		ctx,
+		logger.ServerTypeContextKey, slog.Any(logger.ServerTypeKey, cp.commandServerType.String()),
+	)
+	slog.InfoContext(newCtx, "Closing command plugin")
 
 	cp.subscribeMutex.Lock()
 	if cp.subscribeCancel != nil {
 		cp.subscribeCancel()
 	}
 	cp.subscribeMutex.Unlock()
 
-	return cp.conn.Close(ctx)
+	return cp.conn.Close(newCtx)
 }
 
 func (cp *CommandPlugin) Info() *bus.Info {
 	return &bus.Info{
-		Name: "command",
+		Name: cp.commandServerType.String(),
 	}
 }
 
 func (cp *CommandPlugin) Process(ctx context.Context, msg *bus.Message) {
-	switch msg.Topic {
-	case bus.ConnectionResetTopic:
-		cp.processConnectionReset(ctx, msg)
-	case bus.ResourceUpdateTopic:
-		cp.processResourceUpdate(ctx, msg)
-	case bus.InstanceHealthTopic:
-		cp.processInstanceHealth(ctx, msg)
-	case bus.DataPlaneHealthResponseTopic:
-		cp.processDataPlaneHealth(ctx, msg)
-	case bus.DataPlaneResponseTopic:
-		cp.processDataPlaneResponse(ctx, msg)
-	default:
-		slog.DebugContext(ctx, "Command plugin received unknown topic", "topic", msg.Topic)
+	slog.DebugContext(ctx, "Processing command")
+
+	if logger.ServerType(ctx) == cp.commandServerType.String() || logger.ServerType(ctx) == "" {
```

Comment:
If logger.ServerType(ctx) == "" then we should create a new context with the serverType.